README.md

```diff
@@ -1,32 +1,288 @@
+
+Looking at the objects in memory, we can see there are two memory locations with the label `name`,
+and two memory locations with the label `breed`, etc.  How can we assign the name of the first
+dog to "Bruno" and the name of the second dog to "Penny"?
+
+The solution is to use syntax referred to as "dot notation".  The dot operator `.` is
+used to separate a variable that references an object and the name of a field.  We can
+assign a value to a field for a specific object using dot notation as shown below:
+
+`objectReference.fieldName = value`
+
+To assign the value "Bruno" as the name of the dog referenced by the variable `bigDog`, we write:
+
+`bigDog.name = "Bruno";`
+
+To assign the value "Penny" as the name of the dog referenced by the variable `smallDog`, we write:
+
+`smallDog.name = "Penny";`
+
+Let's update the `main` method to assign values to the fields of each dog
+using the two object references `bigDog` and `smallDog`.
+
+```java
+public class Dog {
+
+   String name;
+   String breed;
+   int age;
+   boolean waggingTail;
+
+   public static void main(String[] args) {
+
+      // create 2 Dog instances
+      Dog bigDog = new Dog();
+      Dog smallDog = new Dog();
+
+      // assign new values to the fields of the first dog
+      bigDog.name = "Bruno";
+      bigDog.breed = "Great Dane";
+      bigDog.age = 8;
+
+      // assign new values to the fields of the second dog
+      smallDog.name = "Penny";
+      smallDog.breed = "Pug";
+      smallDog.age = 5;
+      smallDog.waggingTail = true;
+   }
+}
+```
+
+Notice the `main` method does not assign a value to `bigDog.waggingTail`, so the field
+retains the default boolean value `false`.
+
+Executing the `main` method results in the desired object state:
+
+![dog state assigned values](https://curriculum-content.s3.amazonaws.com/6676/java-mod2-oop-fundamentals/dogs_updated_state.png)
+
+
+## Java Visualizer - Code Along
+
+Currently, the `main` method does not print anything, so it is hard to understand how the state changes as each
+line of code is executed.
+
+Let's step through the `main` method using the IntelliJ debugger and the `Java Visualizer` plugin, which
+will let us see what the objects look like in memory.
+
+1. Set a breakpoint on the first line of executable code in the `main` method (i.e. not a comment or blank line),
+   then click the "debug" icon to launch the debugger.   Your line number may differ depending on comments and blank lines.  
+   
+   ![dog breakpoint](https://curriculum-content.s3.amazonaws.com/6676/java-mod2-oop-fundamentals/set_breakpoint.png)
+
+2. Select the "Java Visualizer" tab in the debug window.
+   Notice the call stack shows the `main` method has stopped at the breakpoint (your line number may be different).    
+ 
+   ![java visualizer tab](https://curriculum-content.s3.amazonaws.com/6676/java-mod2-oop-fundamentals/breakpoint_reached.png)
```

Comment:
Maybe adjust the screenshot with the folders in the project structure collapsed? That way students don't think they need to also have a `Cat` and a `Bike` class?

Reply:
The project name might also throw students?

Reply:
yes, thanks.

Reply:
I updated the image to show a sample project like the student's.